File: demo/java-demo/src/main/java/com/alibaba/testable/demo/service/DemoMockService.java
package com.alibaba.testable.demo.service;

import com.alibaba.testable.demo.model.BlackBox;
import com.alibaba.testable.demo.model.Box;
import org.springframework.stereotype.Service;
import sun.net.www.http.HttpClient;

import java.net.URL;

@Service
public class DemoMockService {

    /**
     * method with new operation
     */
    public String newFunc() {
        BlackBox component = new BlackBox("something");
        return component.get();
    }

    /**
     * method with member method invoke
     */
    public String outerFunc(String s) throws Exception {
        return "{ \"res\": \"" + innerFunc(s) + "\"}";
    }

    /**
     * method with common method invoke
     */
    public String commonFunc() {
        return "anything".trim() + "__" + "anything".substring(1, 2) + "__" + "abc".startsWith("ab");
    }

    /**
     * method with static method invoke
     */
    public BlackBox getBox() {
        return BlackBox.secretBox();
    }

    /**
     * method with override method invoke
     */
    public Box putBox() {
        Box box = new BlackBox("");
        box.put("data");
        return box;
    }

    /**
     * two methods invoke same private method
     */
    public String callerOne() {
        return callFromDifferentMethod();
    }

    public String callerTwo() {
        return callFromDifferentMethod();
    }

    private String innerFunc(String s) throws Exception {
        return HttpClient.New(new URL("http:/xxx/" + s)).getURLFile();
    }

    private String callFromDifferentMethod() {
        return "realOne";
    }

}

File: demo/kotlin-demo/src/main/kotlin/com/alibaba/testable/demo/service/DemoMockService.kt
package com.alibaba.testable.demo.service

import com.alibaba.testable.demo.model.BlackBox
import com.alibaba.testable.demo.model.Box
import com.alibaba.testable.demo.model.ColorBox
import org.springframework.stereotype.Service
import sun.net.www.http.HttpClient
import java.net.URL


@Service
class DemoMockService {

    /**
     * method with new operation
     */
    fun newFunc(): String? {
        return BlackBox("something").get()
    }

    /**
     * method with member method invoke
     */
    fun outerFunc(s: String): String {
        return "{ \"res\": \"" + innerFunc(s) + "\"}"
    }

    /**
     * method with common method invoke
     */
    fun commonFunc(): String {
        val box = BlackBox("anything")
        return box.trim() + "__" + box.substring(1, 2) + "__" + box.startsWith("any")
    }

    /**
     * method with static method invoke
     */
    fun getBox(): BlackBox {
        return ColorBox.createBox("Red", BlackBox.secretBox())
    }

    /**
     * two methods invoke same private method
     */
    fun callerOne(): String {
        return callFromDifferentMethod()
    }

    fun callerTwo(): String {
        return callFromDifferentMethod()
    }

    private fun innerFunc(s: String) = HttpClient.New(URL("http:/xxx/$s")).urlFile

    private fun callFromDifferentMethod() = "realOne"
}
